### PR TITLE
Added form designing guidelines as a separate doc.

### PR DESCRIPTION
**Pull request by mish24**
addresses #20 

## What is included in this PR?
This PR has separate content for [Form designing guidelines](https://opendatakit.org/help/form-design/guidelines/) which is a subset of Form design doc. It has the information and good practices related to creation of forms.

## What is left to be done in the addressed issue?
Add content about:
* Stricter Form Syntax - ODK Aggregate 1.x
* Datastore String Length

Content can be made more brief.

## What problems did you encounter?
Not problems, but verifying the content took more time than anticipated.

### CODE REVIEW

File: form-guidelines.rst [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+Group and Fieldname 
+=====================
+
+Answers to your survey questions are identified by their field names within your form. In ODK Build, the Data Name property specifies the field name. In an XLSForm spreadsheet, the name column specifies the field name. Question groupings also have names, and these guidelines also apply to their names.
+
+Field names appear as the column headers in the Submissions tab in ODK Aggregate. If they are nested within a group, the group name and field name (space-separated) appear as the column header. In the exported CSV file, the group name and field name are concatenated with colon separators (:). Because group names and field names are displayed to the data analyst and/or end user, they should be meaningful.
+
+The structure of the data you send is defined in the ``<instance/>`` section of the XForm. An example, with field names: :guilabel:`name`, :guilabel:`age` and :guilabel:`date`, is shown here:
+
+.. code-block:: html
```
Comment (adammichaelwood) [inline form-guidelines.rst]: xml, not html

File: form-guidelines.rst [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+Compacted names are formed by prefixing the name(s) of the groups in which the field is nested to the field's name, each separated by a ``_`` character. All lower case characters are then capitalized and an ``_`` character is inserted at word breaks; non-alphanumeric characters are also replaced with ``_`` characters. The group prefix is then compacted to consume no more than 1/3 of the column name allowed by the datastore; the field name is compacted to fit the remaining space.
+
+Thus, ``thisFormField`` and ``this-form-field`` will both become ``THIS_FORM_FIELD``, as would a field named Field nested within a thisForm group. Any overlaps of the resulting column names within a single database table are prevented through additional processing steps. In light of these processing steps, it is recommended that field names:
+
+- be short (ideally < 30 characters).
+- must be unique within their enclosing group within the form (this is required by Xml and Javarosa).
```
Comment (adammichaelwood) [inline form-guidelines.rst]: s/Xml/XML

File: form-guidelines.rst [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+.. _forms-with-media:
+
+Forms with media (images, video, audio) included in prompts
+============================================================
+
+You can now use images, audio, and video (in any combination) in addition to or in place of text in questions and in multiple choice answers. To use media we have taken advantage of javarosa's itext framework. Like text, a media file can be different for each language specified in the form.
+
+To define your media, declare one or more ``<text id>`` fields in ``<itext>``. For example:
+
+.. code-block:: html
```
Comment (adammichaelwood) [inline form-guidelines.rst]: xml

File: form-guidelines.rst [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+Video plays in a separate player allowing play, stop, and a scroll bar to jump to a specific spot. Once the video finishes playing, the user will be automatically returned to the form. Users may also return at any time by hitting the phone's "back" button.
+
+.. _usable-media-formats:
+
+Usable Image, Audio and Video Formats
+======================================
+
+The formats listed are what we've tested and verified as working. There may be more formats that are supported by Android.
+
++---------------+---------------------+
```
Comment (adammichaelwood) [inline form-guidelines.rst]: I commend your patience with this ridiculous table syntax.

If you want to make your life easier in the future, check out the `csvtable` directive.

File: form-guidelines.rst [hunk 5]
```diff
@@ -0,0 +1,186 @@
+******************
+Form Guidelines
+******************
+
+.. note::
+
+  This page is a reference guide for users who edit the raw XML in ODK forms. If you use the XLSForm form designer, see http://xlsform.org for form design help.
```
Comment (adammichaelwood) [inline form-guidelines.rst]: I don't know if intro content, which sets the context for understanding the whole doc, belongs in a note.

**Comment by adammichaelwood** [comment]
This is great. 

I made a couple comments. You should also take a line-edit pass on it for typos and things.

----

Let's talk about the title and intro for a second, before we merge it in.

This document isn't precisely about *designing* forms. It's about *coding* (or *building* forms). And more to the point: it is about coding forms in a way that will make them work better specifically with Collect and Aggregate. 

Can you work on the intro paragraph(s) a little and make more clear the purpose of these suggestions?

File: form-guidelines.rst [hunk 6]
```diff
@@ -0,0 +1,186 @@
+******************
+Form Guidelines
```
Comment (adammichaelwood) [inline form-guidelines.rst]: please retitle.

1. XForm, not just form. 
2. Guidelines? What does that mean? Are these suggestions? Best practices? Things that only matter when XForm is used in Collect? 

It is difficult to know at a glance how this relates to the Form Design page in the other PR.

**Comment by mish24** [comment]
@adammichaelwood These are best practices for creating your own forms properly, a sub part of the form design documents.

**Comment by mish24** [comment]
Also, in #239, after binding and body, there are guidelines. So I thought its best to make separate doc for this.

**Comment by adammichaelwood** [comment]
work on the other doc first, then retitle this and make it clear how this is related to that. And address my other edits comments.

**Pull request closed without merging**